src/common.c

```diff
@@ -725,24 +768,30 @@ int fi_epoll_wait(struct fi_epoll *ep, void **contexts, int max_contexts,
 	int i, ret;
 	int found = 0;
 
-	ret = poll(ep->fds, ep->nfds, timeout);
-	if (ret == SOCKET_ERROR)
-		return -ofi_sockerr();
-	else if (ret == 0)
-		return 0;
+	do {
+		ret = poll(ep->fds, ep->nfds, timeout);
+		if (ret == SOCKET_ERROR)
+			return -ofi_sockerr();
+		else if (ret == 0)
+			return 0;
+
+		if (ep->fds[0].revents)
+			fd_signal_reset(&ep->signal);
 
-	for (i = ep->index; i < ep->nfds && found < max_contexts; i++) {
-		if (ep->fds[i].revents) {
-			contexts[found++] = ep->context[i];
-			ep->index = i;
+		for (i = ep->index; i < ep->nfds && found < max_contexts; i++) {
+			if (ep->fds[i].revents) {
+				contexts[found++] = ep->context[i];
+				ep->index = i;
+			}
 		}
-	}
-	for (i = 0; i < ep->index && found < max_contexts; i++) {
-		if (ep->fds[i].revents) {
-			contexts[found++] = ep->context[i];
-			ep->index = i;
+		for (i = 0; i < ep->index && found < max_contexts; i++) {
+			if (ep->fds[i].revents) {
+				contexts[found++] = ep->context[i];
+				ep->index = i;
+			}
 		}
-	}
+	} while (!found);
```

Comment:
Because this loops now, the timeout value needs to be adjusted.  Otherwise, we can end up waiting longer than timeout when an fd has been modified.

Reply:
oh I see. will fix that